### DISCUSSION TITLE
How do i change from Lan/router IP, to my external one?

### DISCUSSION THREAD

**Discussion by 074769** [rejetto/hfs#62] [Q&A] [answered]
How do i change so that others can connect by my real IP and not my lan one, my ports are open, but i cant seem to find an option to switch to my real IP.

**Comment by rejetto** [comment, marked as answer]
you don't *really* need an option in HFS for that.
Indeed, HFS is not responsible for that part, your router is.
Anyway, HFS in the future (hopefully soon) will offer features that will make it easier for you to get the job done, but in the meantime you'll just have to configure your router, and figure out what your external IP is, by yourself.

**Comment by redtrillix** [comment]
@074769 
People can view your file server by going to your external IP, this can be accessed by opening the correct port and ip in your router's firewall and your computer's firewall also known as port forwarding. This is a dangerous way as anyone can access this and can allow for malicious threats through the port, the best solution would be a VPN on your network. 

These are the rough steps for Port Forwarding:
1. Forward your Port on your router (most likely port 80) and your Computer IP (most likely "192.168.x.x").
2. If you are using Windows, you will need to go to "Windows Defender with Advanced Security" and forward the Port or Program.
3. You will then have to find your external IP, which a google search will do: https://www.google.com/search?q=what+is+my+ip, give this to your friends and they should then be able to access the VFS.

**Comment by rejetto** [comment]
I'm no security expert, but I'd like to say that an "open port" is not a security problem by itself.
The possible problem is the software that's responding to that port, if it has bugs or it is not adequately configured, etc.

**Reply by redtrillix** [reply]
This is true, but if you are only trying to open it for friends, VPNs can be a safer solution.

This is what happens when you open ports to the internet, this can be stuff like scanning of ports to vuln testing to known bad IP's attempting access: 
![image](https://user-images.githubusercontent.com/54786587/175263855-8f1ddc22-81fa-4f3e-a051-01e7c714c0cb.png)
Which in my case isn't the worst because I have multiple firewalls, antivirus programs, and IPS, but still not desired.

**Reply by redtrillix** [reply]
I'm not saying you shouldn't port forward, I'm just saying it's not best practice and there are better solutions.

**Reply by rejetto** [reply]
if you want a restricted group of people to access, and you can get them to join your vpn, sure it's good. Not my case, tho.

If my HFS is scanned for wordpress' flaws, the scan will just not be effective and no real harm is done.
Automated attacks don't try to be smart.
In the past HFS 2 has been probed for flaws by some specialists, and some problems has been discovered, and fixed. In that case the real risk was to use an outdated version.
Even if HFS2's bugs had something in common with other software's bugs, an automated attack would fail if not tailored specifically for it.
HFS 3 is totally new, and it would need to be studied independently for flaws, despite me trying my best.

**Reply by 074769** [reply]
That doesn't pose a problem for me, i have router firewall and a software one, plus, i can close/open my ports whenever i want, i can keep 80 port open only for hfs. Anyway, i solved my problem by using another browser to see if i can access it externaly and it works great... Thanks to everyone writing answers. On that topic, it would be great to have an option to change the IP for fast copy/paste action.
